api_v3/services/ThumbAssetService.php

```diff
@@ -793,6 +793,9 @@ function listAction(KalturaAssetFilter $filter = null, KalturaFilterPager $pager
 		}
 		elseif(! $filter instanceof KalturaThumbAssetFilter)
 		{
+		    	if(!is_subclass_of('KalturaThumbAssetFilter', get_class($filter)))
+		        	$filter = $filter->cast('KalturaAssetFilter');
+		    
 			$filter = $filter->cast('KalturaThumbAssetFilter');
```

Comment:
@Galmoss don't you need an else here otherwise you will never create a filter of type "KalturaAssetFilter"